compiler/src/jdk.graal.compiler.test/src/jdk/graal/compiler/core/test/CheckObjectEqualsTest.java

```diff
@@ -0,0 +1,93 @@
+package jdk.graal.compiler.core.test;
+
+import jdk.graal.compiler.core.common.GraalOptions;
+import jdk.graal.compiler.options.OptionValues;
+import jdk.vm.ci.code.InstalledCode;
+import jdk.vm.ci.code.InvalidInstalledCodeException;
+import org.junit.Test;
+
+public class CheckObjectEqualsTest extends GraalCompilerTest {
+
+    public static final Object NULL = null;
+    public static final Object testObject1 = new Object();
+    public static final Object testObject2 = new Object();
+    public static final Object inlineTypeTestObject = new InlineTypeTestClass();
+    public static final OptionValues printProfiling = new OptionValues(getInitialOptions(), GraalOptions.PrintProfilingInformation, true);
+
+    public static value class InlineTypeTestClass {
+    }
+
+    public static boolean snippetNoProfile(Object a, Object b) {
+        return a==b;
+    }
+
+    public static boolean snippetLeftAlwaysNull(Object a, Object b) {
+        return a==b;
+    }
+
+    public static boolean snippetRightAlwaysNull(Object a, Object b) {
+        return a==b;
+    }
+
+    public static boolean snippetLeftInlineType(Object a, Object b) {
+        return a==b;
+    }
+
+    public static boolean snippetRightInlineType(Object a, Object b) {
+        return a==b;
+    }
+
+    @Test
+    public void testNoProfile() throws InvalidInstalledCodeException {
+        resetCache();
+        InstalledCode c = getCode(getResolvedJavaMethod("snippetNoProfile"), null, true, false, getInitialOptions());
+        assert c.executeVarargs(NULL, NULL).equals(true);
+        assert c.isValid();
+    }
+
+    @Test
+    public void testLeftAlwaysNull() throws InvalidInstalledCodeException {
+        resetCache();
+        for (int i = 0; i < 10000; i++) {
+            test("snippetLeftAlwaysNull", NULL, testObject1);
```

Comment:
you can also call test with expected deoptimizations so not sure you need the explicit compile below with the installed code as well

Reply:
u sure? think you can only check if a deoptimization was not triggered. I used CheckCastWithNullTest.java as a template

Reply:
right, only the negative has API, ok then